contracts/contracts/SNStateProofVerifier.sol

```diff
@@ -42,16 +42,17 @@ struct StarknetProof {
 // includes contract proof and state/storage proof for a partciular starknet block
 struct StarknetCompositeStateProof {
     int256 blockNumber;
+    uint256 classCommitment;
     ContractData contractData;
     StarknetProof[] contractProofArray;
     StarknetProof[] storageProofArray;
 }
 
-interface IStarknetResolverService {
-    function addr(bytes32 node)
-        external
-        view
-        returns (StarknetCompositeStateProof memory proof);
+interface PoseidonHash3 {
+    // this is the hades permutation function. TODO update the name when goerli eth is not so expensive
+    function poseidon(
+        uint256[3] calldata input
+    ) external view returns (uint256[3] calldata);
```

Comment:
I'm a bit surprised by `calldata` here. This is only available for function arguments of external functions.

Reply:
I am not so sure about that, you can use it for contract to contract calls and it works fine, its a little bit cheaper I think(but could be wrong) and one should be aware of the downsides(immutability etc etc). But you are right that that is not the convention and probably not worth the confusion it will cause. I have updated it to memory. Interesting question in general though. Happy to be corrected on this understanding